source/extensions/filters/http/transformation/body_header_transformer.cc

```diff
@@ -62,6 +67,97 @@ void BodyHeaderTransformer::transform(
   header_map.setContentLength(body.length());
 }
 
+void BodyHeaderTransformer::parse_headers(
+  const Http::RequestOrResponseHeaderMap &header_map,
+  std::map<std::string, absl::string_view> &headers,
+  std::map<std::string, std::vector<absl::string_view>> &multi_value_headers) const {
+    if (add_request_metadata_) {
+      header_map.iterate(
+        [&headers, &multi_value_headers](const Http::HeaderEntry& header) -> Http::HeaderMap::Iterate {
+            auto header_key = std::string(header.key().getStringView());
+            auto header_value = header.value().getStringView();
+            if (headers[header_key].empty()) {
+                headers[header_key] = header_value;
+            } else {
+                // If there are more than one headers with the same key, use the last one
+                auto existing_header_value = headers[header_key];
+                headers[header_key] = header_value;
+ 
+                // Then, ensure all existing values are in the multiValueHeaders vector
+                if (multi_value_headers[header_key].empty()) {
+                    multi_value_headers[header_key].emplace_back(existing_header_value);
+                }
+
+                multi_value_headers[header_key].emplace_back(header_value);
+            }
+            return Http::HeaderMap::Iterate::Continue;
+        });
+    } else {
+      header_map.iterate(
+      [&headers](const Http::HeaderEntry &header) -> Http::HeaderMap::Iterate {
+        headers[std::string(header.key().getStringView())] =
+            header.value().getStringView();
+        return Http::HeaderMap::Iterate::Continue;
+      });
+    }
+}
+
+void BodyHeaderTransformer::parse_query_string(
+  absl::string_view query_string,
+  std::map<std::string, std::string> &query_string_parameters,
+  std::map<std::string, std::vector<std::string>> &multi_value_query_string_parameters) const {
+    Envoy::Http::Utility::QueryParamsVector query_params = parse_parameters(query_string, 0);
+    for (auto& pair : query_params) {
+      if (query_string_parameters[pair.first].empty()) {
+        query_string_parameters[pair.first] = pair.second;
+      } else {
+        auto existing_value = query_string_parameters[pair.first];
+        query_string_parameters[pair.first] = pair.second;
+        // handle multi value case
+        if (multi_value_query_string_parameters[pair.first].empty()) {
+          multi_value_query_string_parameters[pair.first] = std::vector<std::string>();
+          multi_value_query_string_parameters[pair.first].emplace_back(existing_value);
+        }
+        multi_value_query_string_parameters[pair.first].emplace_back(pair.second);
+      }
+    }
+}
+
+// explain the differences between push_back and emplace_back:
+// the diff
```

Comment:
this looks incomplete?

Reply:
This was a note-to-self that I accidentally committed. I have removed the comment with this commit: https://github.com/solo-io/envoy-gloo/commit/922b07899c3bdbe007ef8e0692a1a02f42b6255d